pkg/workflow/workflowrun/pod.go

```diff
@@ -199,6 +190,33 @@ func (m *PodBuilder) ResolveInputResources() error {
 			return err
 		}
 
+		// Volumed to hold resource data, by default, it's the common PVC in Cyclone, but user can
+		// also specify it in the resource spec.
+		pvc := common.DefaultPvVolumeName
+
+		// Sub-path in the PVC to hold resource data
+		subPath := common.ResourcePath(m.wfr.Name, r.Name)
+
+		// If persistent is set in the resource spec, create a volume for the persistent PVC
+		// specified. Then resource would be pulled in the PVC. If persistent is not set, resource
+		// would be pulled in the common PVC in Cyclone. Note that, data in common PVC would be
+		// cleaned after workflow terminated.
+		persistent := resource.Spec.Persistent
+		if persistent != nil {
+			subPath = filepath.Join(persistent.Path, r.Name)
```

Comment:
Do we really need to join the `r.Name` to the `persistent.Path` configured by users?